src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -90,6 +92,40 @@ public class MessageSender : ClientEntity, IMessageSender
             }
         }
 
+        private async Task<Message> ProcessMessage(Message message)
+        {
+            if (this.RegisteredPlugins == null)
+            {
+                return message;
+            }
+
+            var processedMessage = message;
+            foreach (var plugin in this.RegisteredPlugins)
+            {
+                try
+                {
+                    processedMessage = await plugin.BeforeMessageSend(message).ConfigureAwait(false);
+                }
+                catch (Exception ex) when (ex is ServiceBusPluginException pluginException && pluginException.ShouldCompleteOperation)
```

Comment:
My gut feeling tells me the user. Plugin is there to do the job, not to decide on the overall course of actions.